util/screenshots.js

```diff
@@ -0,0 +1,69 @@
+import fs from "fs";
+import path from "path";
+import * as warcio from "warcio";
+
+
+// ============================================================================
+
+function isScreenshotSelected(params) {
+  return params.screenshot || params.fullPageScreenshot || params.thumbnail;
+}
+
+class Screenshots {
+
+  constructor({page, id, url, date, directory}) {
+    this.page = page;
+    this.id = id;
+    this.url = url;
+    this.directory = directory;
+    this.warcName = path.join(this.directory, "screenshots-" + this.id + ".warc.gz");
+    this.date = date ? date : new Date();
+  }
+
+  async take({fullPage = false, imageType = "png"} = {}) {
+    let options = {omitBackground: true, fullPage: fullPage, type: imageType};
+    let screenshotType = fullPage ? "screenshot-fullpage" : "screenshot";
+    if (imageType === "jpeg") {
```

Comment:
Rather than the logic here, perhaps we can have just an array of settings that encapsulates all the settings:
```
{"screenshot-fullpage": {"viewport", ..., "quality": ..., }
 "thumbnail": ...
}
```
That way, will also be easy to add custom types in the future if needed?
Could just be in this file for now

Reply:
I added discrete options objects for each screenshot type in the constructor for the class, which I think should be nice and extendable. Makes it a bit easier to grab the screenshot type rather than having to access the keys but accomplishes the same result I think. Let me know if you'd rather I change it to use an array.